zkevm-circuits/src/bytecode_circuit/circuit/to_poseidon_hash.rs

```diff
@@ -414,6 +466,30 @@ impl<F: Field, const BYTES_IN_FIELD: usize> ToHashBlockCircuitConfig<F, BYTES_IN
         challenges: &Challenges<Value<F>>,
         fail_fast: bool,
     ) -> Result<(), Error> {
+        // if feature "parallel_syn" is enabled, we will use parallel assignment
+        #[cfg(feature = "parallel_syn")]
+        {
+            // `is_force_serial_assignment` is used for unit test
+            // it's determined by the environment variable `UNIT_TEST_ASSIGNMENT_TYPE`
+            // if enabled, we will bypass the parallel assignment
+            // and use serial assignment instead
+            let assignment_type = std::env::var("UNIT_TEST_ASSIGNMENT_TYPE")
```

Comment:
Maybe `ASSIGNMENT_TYPE` is more appropriate and we need to make this explicit in the PR description so that anyone knows that this PR introduces a new env-var.